docs/release_notes/destinations_v2.js

```diff
@@ -0,0 +1,149 @@
+      return str;
+    } else {
+      return "_" + str;
+    }
+  }
+  function escapeNamespace(namespace) {
+    namespace = convertStreamName(namespace);
+    if (!namespace.charAt(0).match(/[A-Za-z0-9]/)) {
+      namespace = "n" + namespace;
+    }
+    return namespace;
+  }
+
+  function generateSql(namespace, name, raw_dataset) {
+    let v2RawTableName = '`' + bigqueryConvertStreamName(concatenateRawTableName(namespace, name)) + '`';
+    let v1namespace = '`' + escapeNamespace(namespace) + '`';
+    let v1name = '`' + bigqueryConvertStreamName("_airbyte_raw_" + name) + '`';
+    return `CREATE SCHEMA IF NOT EXISTS ${raw_dataset};
+CREATE OR REPLACE TABLE \`${raw_dataset}\`.${v2RawTableName} (
+  _airbyte_raw_id STRING,
+  _airbyte_extracted_at TIMESTAMP,
+  _airbyte_loaded_at TIMESTAMP,
+  _airbyte_data JSON)
+PARTITION BY DATE(_airbyte_extracted_at)
+CLUSTER BY _airbyte_extracted_at
+AS (
+    SELECT
+        _airbyte_ab_id AS _airbyte_raw_id,
+        _airbyte_emitted_at AS _airbyte_extracted_at,
+        CAST(NULL AS TIMESTAMP) AS _airbyte_loaded_at,
+        PARSE_JSON(_airbyte_data) AS _airbyte_data
+    FROM ${v1namespace}.${v1name}
+)`;
+  }
+
+  return (
+    <MigrationGenerator destination="bigquery" generateSql={generateSql}/>
+  );
+}
+
+export const SnowflakeMigrationGenerator = () => {
+  // See SnowflakeSQLNameTransformer
+  function snowflakeConvertStreamName(str) {
+    str = convertStreamName(str);
+    if (str.charAt(0).match(/[A-Za-z_]/)) {
+      return str;
+    } else {
+      return "_" + str;
+    }
+  }
+  function generateSql(namespace, name, raw_schema) {
+    let v2RawTableName = '"' + concatenateRawTableName(namespace, name) + '"';
+    let v1namespace = snowflakeConvertStreamName(namespace);
+    let v1name = snowflakeConvertStreamName("_airbyte_raw_" + name);
+    return `CREATE SCHEMA IF NOT EXISTS "${raw_schema}";
+CREATE OR REPLACE TABLE "${raw_schema}".${v2RawTableName} (
+  "_airbyte_raw_id" STRING NOT NULL PRIMARY KEY,
+  "_airbyte_extracted_at" TIMESTAMP_TZ DEFAULT CURRENT_TIMESTAMP(),
+  "_airbyte_loaded_at" TIMESTAMP_TZ,
+  "_airbyte_data" VARIANT)
+AS (
+    SELECT
+        _airbyte_ab_id AS "_airbyte_raw_id",
+        _airbyte_emitted_at AS "_airbyte_extracted_at",
+        CAST(NULL AS TIMESTAMP_TZ) AS "_airbyte_loaded_at",
+        _airbyte_data AS "_airbyte_data"
+    FROM ${v1namespace}.${v1name}
+)`;
+  }
+  return (
+    <MigrationGenerator destination="snowflake" generateSql={generateSql}/>
+  );
+}
+
+export const MigrationGenerator = ({destination, generateSql}) => {
+  const defaultMessage =
+`Enter your stream's name and namespace to see the SQL output.
+If your stream has no namespace, take the default value from the destination connector's settings.`;
+  const [message, updateMessage] = useState({
+    'message': defaultMessage,
+    'language': 'text'
+  });
+  function updateSql(event) {
+    let namespace = document.getElementById("stream_namespace_" + destination).value;
```

Comment:
There are more react-y ways to reference an element on the page ([e.g.](https://react.dev/learn/referencing-values-with-refs)), and/or this could maintained with [state](https://react.dev/learn/state-a-components-memory).... but these are docs, so YOLO

Reply:
neat! might play with these later but going to merge this as-is for now. from the docs - it sounds like using state is the more correct way to do it? since I _do_ want to rerender when the text fields change

unless I'm just misunderstanding what rendering even means :shrug: